ompi/mpi/fortran/mpif-h/session_get_pset_info_f.c

```diff
@@ -96,6 +96,8 @@ void ompi_session_get_pset_info_f(MPI_Fint *session,char *pset_name, MPI_Fint *i
     c_ierr = PMPI_Session_get_pset_info(c_session, c_name, &c_info);
     if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
 
+    free (c_name);
```

Comment:
Do you want to protect this `free (c_name)` with a `if (NULL != c_name)` ?

Reply:
We used to care about that, but it's allowable in POSIX to `free(NULL)`.  So while I think we stopped caring about that a while ago, there's still a bunch of places in the code base that go both ways (some check for NULL, some don't).